### ISSUE TITLE
BuddyPress/BBPress errors

### ISSUE THREAD

**Issue by misiman** [maximeschoeni/sublanguage#17]
Hi,

Your plugin looks like it could be really perfect for my requirements, lightweight and with inheritance.

However, I am testing the latest version (2?) from the Github repo and have found it conflicts with BuddyPress and/or BBPress with the following errors produced in the Admin console:

1) Notice: Trying to get property of non-object in C:\Websites\www.example.dev\wp-content\plugins\bbpress\includes\common\functions.php on line 1446 which appears to be related to 	bbp_query_post_parent__in( )

2) Warning: in_array() expects parameter 2 to be array, null given in C:\Websites\www.example.dev\wp-content\plugins\bbpress\includes\common\functions.php on line 1446 which also appears to be related to bbp_query_post_parent__in( ) but specifically the call to in_array ( )

It would be great to resolve these issues because BuddyPress and BBPress are commonly used plugins.

Thanks

**Comment by maximeschoeni** [comment]
Hello, thank you for your report, I will check this. The 2.0 version is not stable yet but it should be soon. I am actually trying to foresee issues it may have with other plugins.

**Comment by maximeschoeni** [comment]
I think this is a problem in BBPress. I submitted a [fix](https://github.com/wp-plugins/bbpress/pull/2/commits/197dbd2a1220d4937edaf367ff7ee6bf1c22c409). Hope they will patch it!

**Comment by misiman** [comment]
Nice detective work.

Do you know why the BBPress issue only presents itself after activating Sublanguage? I can see the problem you highlight, I would just like to understand the logic behind why the conflict doesn't manifest itself with other plugins I've used.

Many thanks.

**Comment by maximeschoeni** [comment]
This conflict happens every time a plugin needs to query database (using WP_Query class) before a certain point. Sublanguage needs to fetch the language items very soon, that's when the error is triggered. Your other plugins don't need to do query so early, that's why they don't trigger it.

Alternatively I could also avoid using WP_Query class and use a raw SQL query instead. On second look it doesn't sound bad because it would also prevent other conflicts... I'll definitely consider it. Thank you for submitting this issue!

**Comment by misiman** [comment]
Thanks for the explanation - a well sanitised raw SQL query sounds like the best option.

Do you know when the earliest v2 RC will be available?

**Comment by maximeschoeni** [comment]
The main drawback of this approach is about cache: a raw sql query does not benefit from a cache plugins in case there is one.

I hope it will be ready next week.

**Comment by misiman** [comment]
Hi again,

The new plugin looks really good. Am I right in thinking that post translations are now stored in post_meta rather than wp_posts?

Does this cause a problem for the standard Wordpress mechanisms for free-text searching post content?

Thanks

**Comment by maximeschoeni** [comment]
Yes, you're right, from 2.0, translations are stored in post_meta. It is a less hacky way of doing and it improves general compatibility. But as you point, it also cause problem with the wordpress search system. I am aware of this and I am looking for a fix.

**Comment by misiman** [comment]
Ok, thanks.

This is an important feature for me (and I expect for most others using translation), because free-text search in any supported language locale is pretty essential to a good user experience.
